src/js/app/app.js

```diff
@@ -0,0 +1,66 @@
+// eslint-disable-next-line import/no-mutable-exports
+import $ from 'js/dom/dom';
+import 'js/modalToggle';
+import youtubeSearch from 'js/youtubeSearch';
+import youtubeArticle from 'js/components/youtubeArticle';
+
+// global state
+let playListState = {
+  // will, watched, like 세가지 상태를 다룰 예정
+  status: 'will',
+  list: JSON.parse(localStorage.getItem('playList'))?.list ?? [],
+};
+
+const setPlayListState = newPlayListState => {
+  playListState = newPlayListState;
+};
+
+$.playListLength.textContent = playListState.list.length;
+
+// 플레이리스트를 업데이트 시켜 줄 render 함수
+const render = newPlayList => {
+  if (newPlayList) setPlayListState({ ...playListState, list: newPlayList });
+  $.mainVideoWrapper.innerHTML = '';
+  if (playListState.list.length) {
+    const $fragment = document.createDocumentFragment();
+    playListState.list.forEach(item => {
+      if (playListState.status === 'will') {
+        if (!item.isWatch) {
+          $fragment.appendChild(youtubeArticle(item, playListState, render));
+        }
+      } else if (playListState.status === 'watched') {
+        if (item.isWatch) {
+          $fragment.appendChild(youtubeArticle(item, playListState, render));
+        }
+      }
+    });
+    $.mainVideoWrapper.appendChild($fragment);
+  } else {
+    $.emptyMessageContainer.classList.add('empty');
+  }
```

Comment:
코드의 Depth가 깊어질수록 유지보수 하기가 힘들어진답니다 ㅠㅠ
그래서
```suggestion
  if (playListState.list.length === 0) {
    return  $.emptyMessageContainer.classList.add('empty');
  }

  const $fragment = document.createDocumentFragment();
  
  playListState.list.forEach(item => {
    if (
      (playListState.status === 'will' && !item.isWatch) ||
      (playListState.status === 'watched') && item.isWatch)
    ) 
    $fragment.appendChild(youtubeArticle(item, playListState, render));
  });
  $.mainVideoWrapper.appendChild($fragment);
```
이런식으로 표현해주면 어떨까요?